bindings/go/plugin/manager/registries/componentlister/registry.go

```
@@ -0,0 +1,254 @@
+		Plugin: listerPlugin,
+		cmd:    plugin.Cmd,
+	}
+
+	// wrap the untyped internal plugin into a typed representation.
+	return listerPlugin, nil
+}
+
+// GetComponentListerCredentialConsumerIdentity retrieves the consumer identity
+// for a component lister based on a given repository specification.
+func (r *ComponentListerRegistry) GetComponentListerCredentialConsumerIdentity(ctx context.Context,
+	repositorySpecification runtime.Typed,
+) (runtime.Identity, error) {
+	r.mu.Lock()
+	defer r.mu.Unlock()
+
+	// Check if this is an internal plugin first
+	typ := repositorySpecification.GetType()
+	if ok := r.scheme.IsRegistered(typ); ok {
+		p, ok := r.internalComponentListerPlugins[typ]
+		if !ok {
+			return nil, fmt.Errorf("no internal plugin registered for type %v", typ)
+		}
+
+		identity, err := p.GetComponentListerCredentialConsumerIdentity(ctx, repositorySpecification)
+		if err != nil {
+			return nil, fmt.Errorf("failed to get identity for the given specification '%+v': %w", repositorySpecification, err)
+		}
+
+		return identity, nil
+	}
+
+	// For external plugins, get the plugin and ask for identity
+	plugin, err := r.getPlugin(ctx, typ)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get plugin for typ %q: %w", typ, err)
+	}
+
+	request := &v1.GetIdentityRequest[runtime.Typed]{
+		Typ: repositorySpecification,
+	}
+
+	result, err := plugin.GetIdentity(ctx, request)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get identity: %w", err)
+	}
+
+	return result.Identity, nil
+}
+
+// GetComponentLister returns ComponentLister for a specific repository type.
+func (r *ComponentListerRegistry) GetComponentLister(ctx context.Context,
+	repositorySpecification runtime.Typed,
+	credentials map[string]string,
+) (repository.ComponentLister, error) {
+	r.mu.Lock()
+	defer r.mu.Unlock()
+
+	// look for an internal implementation that actually implements the interface
+	if _, err := r.scheme.DefaultType(repositorySpecification); err != nil {
```

this should not error.

---

Ok, ignoring the error with https://github.com/open-component-model/open-component-model/pull/993/commits/b5bfcbabfffe350bfa8aebae2464c0c849ffacdf

p.s. I'd be interested to see an example illustrating the problem with the error. Didn't get my head around it yet.